pkg/networkserver/grpc_asns.go

```diff
@@ -81,16 +81,18 @@ func (ns *NetworkServer) LinkApplication(link ttnpb.AsNs_LinkApplicationServer)
 
 	logger.Debug("Linked application")
 
-	events.Publish(evtBeginApplicationLink(ctx, ids, nil))
-	defer events.Publish(evtEndApplicationLink(ctx, ids, err))
+	go events.Publish(evtBeginApplicationLink(ctx, ids, nil))
```

Comment:
@htdvisser is there a reason (not) to publish events in goroutines?

If there's a good reason to do that, should `Publish()` be non-blocking (if it's not)?

I don't like adding subtleties like this in code that are for no obvious reason.

Reply:
There is no reason to publish events from goroutines. The default `events.PubSub` publishes to a buffered channel (default size 64) and `Publish` doesn't block unless that channel is full.

Reply:
Also not doing it with `go events.Publish(...)` ensures that the order of the events is maintained (Otherwise publishing an End event could occur before publishing the Start event - the channel maintains this order).